test.md

```diff
@@ -258,6 +295,35 @@ We allow to `invoke` a function by its exported name in the test code.
          <exports> ... ENAME |-> TFIDX ... </exports>
 ```
 
+### Registering Modules
+
+We first define how modules get stored away.
+
+```k
+    rule <k> #storeInstance => . ... </k>
+         <moduleInst> INST </moduleInst>
+         <nextModuleIdx> NEXT => NEXT +Int 1 </nextModuleIdx>
+         <moduleInstances> ... .Map => NEXT |-> INST ... </moduleInstances>
+```
+
+We will reference modules by name in imports.
+`register` is the instruction that allows us to associate a name with a module.
+
+```k
+    syntax Auxil ::= "(" "register" String            ")"
+                   | "(" "register" String Identifier ")"
+                   |     "register" String Int
+ // ------------------------------------------
+    rule <k> ( register S ) => register S (NEXT -Int 1) ... </k> // Register last module.
+         <nextModuleIdx> NEXT </nextModuleIdx>
+         requires NEXT >Int 0
+    rule <k> ( register S ID:Identifier ) => register S IDX ... </k>
+         <moduleIds> ... ID |-> IDX ... </moduleIds>
+    rule <k> register S:String IDX:Int => . ... </k>
+         <moduleInstances> ... IDX |-> INST ... </moduleInstances>
+         <moduleRegistry> ... .Map => S |-> INST ... </moduleRegistry>
```

Comment:
Do we really need to copy INST? Can we just store the IDX here?

Reply:
Absolutely right. Actually, before I pushed the fix for this, the tests were failing.